provision/kubernetes/deploy.go

```diff
@@ -1254,6 +1254,27 @@ func (m *serviceManager) DeployService(ctx context.Context, opts servicecommon.D
 	if err != nil {
 		return errors.WithStack(err)
 	}
+
+	if opts.OverrideVersions {
+		var deps *appsv1.DeploymentList
```

Comment:
It's more safe defining it as a regular variable instead of a pointer.
```suggestion
		var deps appsv1.DeploymentList
```